oauth2withjwt/src/main/java/com/brother/oauth2withjwt/security/CustomAuthenticationManager.java

```diff
@@ -0,0 +1,45 @@
+package com.brother.oauth2withjwt.security;
+
+import com.brother.oauth2withjwt.entity.UserEntity;
+import com.brother.oauth2withjwt.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CustomAuthenticationManager implements AuthenticationManager {
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    PasswordEncoder passwordEncoder;
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        String username = authentication.getPrincipal() + "";
+        String password = authentication.getCredentials() + "";
+
+        UserEntity userEntity = userRepository
```

Comment:
If we used Optional in the Repository layer. then we use orElseTrhow() in the place of null check

Reply:
done